pages/graveyard/[id].js

```diff
@@ -0,0 +1,96 @@
+import { useRouter } from "next/router";
+import SVGIcon from "@/components/SVGIcon";
+import { formatPetsAge } from "@/utils";
+import StyledLink from "@/components/Styles/StyledLink";
+import StyledHeading from "@/components/Styles/StyledHeading";
+import styled from "styled-components";
+
+export default function PetsDetail({ deceasedPets }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (!deceasedPets) {
+    return <div>...loading</div>;
+  }
+
+  const detailsPet = deceasedPets.find((pet) => pet.id === id);
+
+  if (!detailsPet) {
+    return <div>...loading</div>;
+  }
```

Comment:
What if you always get !deceasedPets or !detailsPet - in this case the user will only see ...loading but maybe there is no data to be loaded...